CriticalEvents events
MovedGaps gap_moves